.github/workflows/test.yaml

```diff
@@ -25,19 +25,29 @@ on:
 
 jobs:
   test:
-    # NOTE: The worker environment running this job needs firefox and
-    #       geckowebdriver available. Before upgrading to ubuntu-24.04 in the
-    #       future, check and see its available in a location like this:
-    #       https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#browsers-and-drivers
-    #
-    runs-on: ubuntu-22.04
-
+    timeout-minutes: 30
```

Comment:
this seems... like a lot... is it still needed? could individual offending tests be constrained with e.g. `pytest-timeout`?

Reply:
The tests seems to be taking up to 17 minutes, so this is roughly double to allow the occasional slow test run. previously there was no time-limit.